Consume renamed TargetFramework package

ViktorHofer opened the pull request:
Depends on https://github.com/dotnet/arcade/pull/8421

ViktorHofer committed:
Consume renamed TargetFramework package

ViktorHofer commented:
@safern @akoeplinger do you know if these failures are caused by my changes?

- Helix API does not contain an entry for OSX.1014.Amd64.Open
- Helix API does not contain an entry for Windows.10.Amd64.Client19H1.Ope

cc @dotnet/dnceng

jonfortescue commented:
@ViktorHofer Not caused by your changes. Those queues were both deleted by the latest rollout due to the operating systems reaching end of life.

garath commented:
> Those queues were both deleted by the latest rollout due to the operating systems reaching end of life.

I believe the change for the OSX queue is happening in #64565

safern commented:
The windows one is: https://github.com/dotnet/runtime/pull/64699

ViktorHofer commented:
Thanks all. I'll give this another try tomorrow. Meanwhile @safern or @ericstj would you mind reviewing and approving?

safern commented on src/libraries/Microsoft.NETCore.Platforms/src/Microsoft.NETCore.Platforms.csproj:
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppToolCurrent);$(NetFrameworkToolCurrent)</TargetFrameworks>
+    <UseTargetFrameworkSDK>false</UseTargetFrameworkSDK>
```
Since the package was rename to not have an "SDK" suffix, I would rename this to something like, `UseTargetFrameworkBuildTasks`?

Deeprai32037i replied:
Someone hacked my personal information my whole personal data my old Gmail account my old Samsung account my all Google Drive data on my Google photos Michael all other information like Facebook messenger telegram Instagram to talk to either Snapchat WhatsApp someone hacked what's up big dog is very important because there's access not have and Instagram creator into much pain dirty up and development my Facebook on Apple developing and that my iPhone and my mama Gmail account my brother Gmail account to my house IP address and my Internet IP address my devices IP address and create you are a numberMy name on the two much create on the domain name and use someone of my view my YouTube channel something my TV hacker anything to hurt you can help me please my iCloud iPhone or your iPad or laptop anything ahead you can help me plea

akoeplinger commented:
@ViktorHofer looks like this runs into the same issue as the darc update PR, do you know why the System.Private.CoreLib.Generators.dll ends up in CORE_ROOT:  https://github.com/dotnet/runtime/pull/64376#issuecomment-1028521765 ?

ViktorHofer commented:
@akoeplinger the NativeAot leg on this PR isn't failing like in https://github.com/dotnet/runtime/pull/64376#issuecomment-1028521765, or am I overlooking something?

akoeplinger commented:
@ViktorHofer yeah that's because https://github.com/dotnet/runtime/pull/64715 added some handling so the nativeaot doesn't crash, but the underlying issue is still there (and affects the Mono llvm legs)

ViktorHofer commented:
> and affects the Mono llvm legs

So the Mono llvm legs should fail on my PR? If not, how can I check if the issue is happening on my PR?

ViktorHofer commented:
Based on the time this started failing it could be https://github.com/dotnet/arcade/commit/93656ab81d1b7f4507edc971465ef2e640694e64.

ViktorHofer commented:
OK I have a lead.

ViktorHofer committed:
Rename CoreLibs generator parent dir

Projects under src/libraries/ which are located under a "gen" directory
are automatically treated as source generator projects. Because the
CoreLib generator was placed under a different directory "generators",
it was treated as a RuntimeAssembly which gets binplaced into the
runtime path. The runtime tests then take the assemblies from there and
copy them into their CORE_ROOT directory.

Renaming the CoreLib source generators parent directory fixes that so
that it is treated as source generator and it brings consistency into
src/libraries.

ViktorHofer commented:
@akoeplinger the fix is in the second commit. The actual issue existed since the CoreLib source generator was added to src/libraries. In https://github.com/dotnet/runtime/commit/9580b7dbe54c52a8b977ed4bb5819f43d6eb5e93 @ericstj disabled binplacing source generators into the runtime path via a convention but CoreLib's source generator didn't comply to that convention.

This popped up with my change now because binplacing didn't work for non multi-targeting projects. Only projects with a `<TargetFrameworks />` property were chosen as a binplace candidate and projects with a `<TargetFramework />` property were ignored. I fixed that in https://github.com/dotnet/arcade/commit/93656ab81d1b7f4507edc971465ef2e640694e64 and hence the CoreLib generator started to fail because it was the first generator to not multi-target.

Would appreciate an approval of the second commit (it's just a simple rename). The first commit is already approved.

MichalStrehovsky commented:
> @akoeplinger the fix is in the second commit. The actual issue existed since the CoreLib source generator was added to src/libraries. In https://github.com/dotnet/runtime/commit/9580b7dbe54c52a8b977ed4bb5819f43d6eb5e93 @ericstj disabled binplacing source generators into the runtime path via a convention but CoreLib's source generator didn't comply to that convention as its parent directory was named "generators" instead of "gen".

Ah, so indeed similar to #62836. I was puzzled what it would be a problem just now. I can sign off on the second commit.

MichalStrehovsky reviewed (approved):
Second commit LGTM

ViktorHofer commented:
More than 3 hours for PR validation, that's crazy :O

premun commented:
> In total more than 3 hours for PR validation, that's crazy :O

_something something_ mono repo _something something_ 😅

marek-safar commented:
> In total more than 3 hours for PR validation, that's crazy :O

/cc @jkotas

ericstj commented:
> @akoeplinger Alexander Köplinger FTE the fix is in the second commit. The actual issue existed since the CoreLib source generator was added to src/libraries. In https://github.com/dotnet/runtime/commit/9580b7dbe54c52a8b977ed4bb5819f43d6eb5e93 @ericstj Eric St. John FTE disabled binplacing source generators into the runtime path via a convention but CoreLib's source generator didn't comply to that convention as its parent directory was named "generators" instead of "gen".

Perhaps we should have stronger rules around project classification.  IOW: make sure projects fall into a known convention, specify what they are, otherwise it's an error.

ericstj commented:
> In total more than 3 hours for PR validation, that's crazy :O

Seems this is the long-pole.

> CoreCLR Pri0 Runtime Tests Run windows arm64 checked
> Duration: 2h 41m 22s

https://dev.azure.com/dnceng/public/_build/results?buildId=1589294&view=logs&j=bb94f37e-a09b-5b51-09b2-ab78e7aa5103

ViktorHofer commented:
> CoreCLR Pri0 Runtime Tests Run windows arm64 checked
Duration: 2h 41m 22s

That leg doesn't start immediately as it depends on both the coreclr and libraries product build. Add 20 minutes to it and you are over the 3 hour mark. Another long pole was the llvmfullaot leg.

MattGal commented:
> > CoreCLR Pri0 Runtime Tests Run windows arm64 checked
> > Duration: 2h 41m 22s
> 
> That leg doesn't start immediately as it depends on both the coreclr and libraries product build. Add 20 minutes to it and you are over the 3 hour mark.
> 
> Another long pole is the llvmfullaot leg which in total also takes more than 3 hours.

Just to pile on, since this thread keeps popping up and I had to hit unsubscribe, is that this is 4 days, 19 hours of machine time in Helix tests for the last build alone:
``` 
let JobIds = 
Jobs 
| where Source == "pr/public/dotnet/runtime/refs/pull/64670/merge" 
| extend props = parse_json(Properties)
| where props["BuildNumber"]=="20220203.14"
| project JobId;
WorkItems
| where JobId  in (JobIds)
| where Name != JobName
| extend runtime = Finished-Started
| summarize sum(runtime)
``` 
(gives 4.19:50:06.7330000)

jkotas commented:
> CoreCLR Pri0 Runtime Tests Run windows arm64 checked
> Duration: 2h 41m 22s

It looks like we have hit low Win Arm64 Helix availability. Can we tell what caused it?

In past week, this job executed in 84 minutes on average. There are jobs running for longer than this job or consuming more machine hours that this job. For example, the `Mono llvmfullaot Pri0 Runtime Tests Run  Linux arm64 release` tool 142 minutes on average last week.

> this is 4 days, 19 hours of machine time in Helix tests for the last build alone:

Yes, we know that each CI run in dotnet/runtime consumes days worth of build machine time and days worth of test machine time. 

The lowest hanging fruit to improve CI machine costs is to eliminate idle build machines waiting for Helix jobs to finish. Tracked by https://github.com/dotnet/arcade/issues/7068
